src/Sylius/Bundle/CoreBundle/EventListener/ChannelListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Resource\Event\ResourceEvent;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+
+/*
+ * Generic channel listener.
+ *
+ * @author Gustavo Perdomo <gperdomor@gmail.com>
+ */
+class ChannelListener
```

Comment:
@pjedrzejewski @pamil mmmmmm... I think <code>ChannelDeletionListener</code> is a better name